### ISSUE TITLE
Custom radio buttons example does not work in TypeScript

### ISSUE THREAD

**Issue by rossng** [chakra-ui/chakra-ui#4089]
# 🐛 Bug report

I'm trying to use the Chakra UI [custom radio buttons example](https://chakra-ui.com/docs/form/radio#custom-radio-buttons) for my TypeScript React project.

## 💥 Steps to reproduce

Copy the example component into your project, adding the necessary props type:

```
function RadioCard(props: PropsWithChildren<UseRadioProps>) {
    const { getInputProps, getCheckboxProps } = useRadio(props);

    const input = getInputProps();
    const checkbox = getCheckboxProps();

    return (
        <Box as="label">
            <input {...input} />
            <Box
                {...checkbox}
                cursor="pointer"
                borderWidth="1px"
                borderRadius="md"
                boxShadow="md"
                _checked={{
                    bg: "teal.600",
                    color: "white",
                    borderColor: "teal.600",
                }}
                _focus={{
                    boxShadow: "outline",
                }}
                px={5}
                py={3}
            >
                {props.children}
            </Box>
        </Box>
    );
}
```

Experience the following compilation error:

```
error TS2322: Type '{ children: ReactNode; cursor: "pointer"; borderWidth: "1px"; borderRadius: "md"; boxShadow: "md"; _checked: 
{ bg: string; color: string; borderColor: string; }; _focus: { boxShadow: string; }; ... 256 more ...; css?: CSSProp<...> | undefined; }' is not assignable to type 'ChakraProps'.
    Types of property 'css' are incompatible.
      Type 'CSSProp<any> | undefined' is not assignable to type 'Interpolation<{}>'.
        Type 'FlattenInterpolation<ThemeProps<any>>' is not assignable to type 'Interpolation<{}>'.
          Type 'readonly Interpolation<ThemeProps<any>>[]' is missing the following properties from type 'ArrayInterpolation<{}>': pop, push, reverse, shift, and 6 more.
```

## 🧐 Expected behavior

The example should work in a TypeScript project.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |  1.6.1  |
| Browser          |  N/A  |
| Operating System | Windows |

**Comment by mjgwood** [comment]
Not sure if I should make a separate issue since I'm getting a different error, but the [Custom Radio Buttons example](https://chakra-ui.com/docs/form/radio#custom-radio-buttons) throws a TypeScript error for me as well.

The issue is with `const radio = getRadioProps({ value });`:

```
Argument of type '{ value: string; }' is not assignable to parameter of type 'Merge<HTMLProps<HTMLInputElement>, { onChange?: ((e: EventOrValue) => void) | undefined; value?: string | number | undefined; } & Pick<InputHTMLAttributes<HTMLInputElement>, "color" | ... 281 more ... | "type">>'.
  Property 'enterKeyHint' is missing in type '{ value: string; }' but required in type 'Pick<InputHTMLAttributes<HTMLInputElement>, "color" | "width" | "height" | "defaultChecked" | "defaultValue" | "suppressContentEditableWarning" | ... 276 more ... | "type">'.ts(2345)
```

Passing the `enterKeyHint` property to `getRadioProps()` resolves the error, but this is not explained in the docs example, and I'm not sure why this property is required.

**Comment by CaitlinSweeney** [comment]
I think I had a similar error @mjgwood. Did you try using the `UseRadioProps`?

Also, check to make sure your `value` is a string, ie: `{...getRadioProps({ value: someValue.toString() })}`

**Comment by segunadebayo** [comment]
Hi @rossng,

This error is very common when you have multiple instances of emotion in `node_modules`. Storybook is usually the suspect here since they use the older versions of emotion which pollutes the TS namespace with that `css` type definition.

I'd recommend creating a `.yarnclean` file and include `@emotion/core/types` to fix that type definition. You can so try @CaitlinSweeney's solution (thank you, Caitlin!)

If the issue persists, please send a reproduction link.

**Comment by cabello** [comment]
Hello 👋 
I just got blocked by this, my app does not load and errors out:

```typescript
/app/src/@venue/features/spaces/SpacesContainer.tsx
TypeScript error in /app/src/@venue/features/spaces/SpacesContainer.tsx(174,10):
Type '{ children: string | ReactElement<any, string | JSXElementConstructor<any>>; slot?: string; style?: CSSProperties; title?: string; ... 251 more ...; css?: InterpolationWithTheme<...>; }' is not assignable to type 'ChakraProps'.
  Types of property 'css' are incompatible.
    Type 'InterpolationWithTheme<any>' is not assignable to type 'Interpolation<{}>'.
      Type 'ArrayInterpolation<undefined>' is not assignable to type 'Interpolation<{}>'.
        Type 'ArrayInterpolation<undefined>' is not assignable to type 'ArrayInterpolation<{}>'.
          The types returned by 'pop()' are incompatible between these types.
            Type 'Interpolation<undefined>' is not assignable to type 'Interpolation<{}>'.
              Type 'ArrayInterpolation<undefined>' is not assignable to type 'Interpolation<{}>'.  TS2322

    172 |       <Box as='label'>
    173 |         <input {...getInputProps()} />
  > 174 |         <StyledSpaceBox {...getCheckboxProps()}>
        |          ^
    175 |           {props.children}
    176 |         </StyledSpaceBox>
    177 |       </Box>
```

The component part that I think it matters has the following:

```typescript
import { Box, useRadio, UseRadioProps } from '@chakra-ui/react';

interface ExtendedUseRadioProps extends UseRadioProps {
  children: React.ReactElement | string;
}

const RadioCheck = (props: ExtendedUseRadioProps): React.ReactElement => {
  const { getInputProps, getCheckboxProps } = useRadio(props);
  // ...

  return (
        <StyledSpaceBox {...getCheckboxProps()}>
          {props.children}
        </StyledSpaceBox>
  );
}

const StyledSpaceBox = styled(Box)`
  cursor: pointer;
`;
```

My `package.json` has the following:

```javascript
"@chakra-ui/react": "^1.6.4",
"@emotion/css": "^11.0",
"@emotion/react": "^11.0",
"@emotion/styled": "^11.0",
"@storybook/addon-actions": "^6.4.0-beta.15",
"@storybook/addon-essentials": "^6.4.0-beta.15",
"@storybook/addon-links": "^6.4.0-beta.15",
"@storybook/addons": "^6.4.0-beta.15",
"@storybook/node-logger": "^6.4.0-beta.15",
"@storybook/preset-create-react-app": "^3.2.0",
"@storybook/react": "^6.4.0-beta.15",
```

Any suggestions on how to fix this? I did try adding `@emotion/core/types` to `.yarnclean`, this file did not exist in my system so I just created with a single line, but it did not work.

**Comment by cabello** [comment]
Because this was not happening before, I did a `git bisect` to find which change introduced the issue, after inspecting a `yarn.lock` I found out that `csstype 3.0.9` was causing the issue, here is my diff:

```diff
diff --git a/yarn.lock b/yarn.lock
index a5a20ddb..47787617 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -9475,12 +9475,7 @@ csstype@^2.5.7:
   resolved "https://registry.yarnpkg.com/csstype/-/csstype-2.6.16.tgz#544d69f547013b85a40d15bff75db38f34fe9c39"
   integrity sha512-61FBWoDHp/gRtsoDkq/B1nWrCUG/ok1E3tUrcNbZjsE9Cxd9yzUirjS3+nAATB8U4cTtaQmAHbNndoFz5L6C9Q==
 
-csstype@^3.0.2:
-  version "3.0.9"
-  resolved "https://registry.yarnpkg.com/csstype/-/csstype-3.0.9.tgz#6410af31b26bd0520933d02cbc64fce9ce3fbf0b"
-  integrity sha512-rpw6JPxK6Rfg1zLOYCSwle2GFOOsnjmDYDaBwEcwoOg4qlsIVCN789VkBZDJAGi4T07gI4YSutR43t9Zz4Lzuw==
-
-csstype@^3.0.6:
+csstype@^3.0.2, csstype@^3.0.6:
   version "3.0.8"
   resolved "https://registry.yarnpkg.com/csstype/-/csstype-3.0.8.tgz#d2266a792729fb227cd216fb572f43728e1ad340"
   integrity sha512-jXKhWqXPmlUeoQnF/EhTtTl4C9SnrxSH/jZUih3jmO6lBKr99rP3/+FmrMj4EFpOXzMtXHAZkd3x0E6h6Fgflw==
```

Hopefully this will be useful to somebody out there in the world wide web. 🤓

**Comment by leanhvu21042001** [comment]
> Because this was not happening before, I did a `git bisect` to find which change introduced the issue, after inspecting a `yarn.lock` I found out that `csstype 3.0.9` was causing the issue, here is my diff:
> 
> ```diff
> diff --git a/yarn.lock b/yarn.lock
> index a5a20ddb..47787617 100644
> --- a/yarn.lock
> +++ b/yarn.lock
> @@ -9475,12 +9475,7 @@ csstype@^2.5.7:
>    resolved "https://registry.yarnpkg.com/csstype/-/csstype-2.6.16.tgz#544d69f547013b85a40d15bff75db38f34fe9c39"
>    integrity sha512-61FBWoDHp/gRtsoDkq/B1nWrCUG/ok1E3tUrcNbZjsE9Cxd9yzUirjS3+nAATB8U4cTtaQmAHbNndoFz5L6C9Q==
>  
> -csstype@^3.0.2:
> -  version "3.0.9"
> -  resolved "https://registry.yarnpkg.com/csstype/-/csstype-3.0.9.tgz#6410af31b26bd0520933d02cbc64fce9ce3fbf0b"
> -  integrity sha512-rpw6JPxK6Rfg1zLOYCSwle2GFOOsnjmDYDaBwEcwoOg4qlsIVCN789VkBZDJAGi4T07gI4YSutR43t9Zz4Lzuw==
> -
> -csstype@^3.0.6:
> +csstype@^3.0.2, csstype@^3.0.6:
>    version "3.0.8"
>    resolved "https://registry.yarnpkg.com/csstype/-/csstype-3.0.8.tgz#d2266a792729fb227cd216fb572f43728e1ad340"
>    integrity sha512-jXKhWqXPmlUeoQnF/EhTtTl4C9SnrxSH/jZUih3jmO6lBKr99rP3/+FmrMj4EFpOXzMtXHAZkd3x0E6h6Fgflw==
> ```
> 
> Hopefully this will be useful to somebody out there in the world wide web. 🤓

Hi @cabello, how did you fix issue?
